### DISCUSSION TITLE
Question: RemoteRunnable and tools - what am I missing?

### DISCUSSION THREAD

**Discussion by itamarhaber** [langchain-ai/langserve#559] [Q&A]
Hi everyone,

I have a question that either makes sense or doesn't - would appreciate any guidance.

Context: I want to implement an agent that can use a tool. The agent, alongside the different API keys, should be hosted and served with langserve. The tool, on the other hand, needs to be executed by the agent's caller.

How would one go about that?

**Comment by eyurtsev** [comment]
The default agent executor in langchain does not allow doing unfortunately -- it hard coded the assumption that tool execution happens in the same process. 

I'd look into writing a custom agent executor using langgraph: https://python.langchain.com/docs/langgraph

Similar to question: https://github.com/langchain-ai/langserve/discussions/334
